### DISCUSSION TITLE
raspicam_node

### DISCUSSION THREAD

**Discussion by questrov** [fjp/2wd-robot#52] [Q&A] [answered]
[Ubiquity Robotics raspicam node](https://github.com/UbiquityRobotics/raspicam_node) seems to be for Kinetic.  Tried installing from source but get the following error:

`Err:5 https://packages.ubiquityrobotics.com/ubuntu/ubiquity xenial InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 4BBE95A1C3032ED8
Reading package lists... Done
W: GPG error: https://packages.ubiquityrobotics.com/ubuntu/ubiquity xenial InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 4BBE95A1C3032ED8
E: The repository 'https://packages.ubiquityrobotics.com/ubuntu/ubiquity xenial InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
`

Poking around the various threads seems to be a rabbit hole.  I've run both raspicam_node and [usb_cam](http://wiki.ros.org/usb_cam) playing around with with the stock [Ubiquity Robotics Raspberry Pi Images](https://downloads.ubiquityrobotics.com/pi.html) but have yet to try it on Noetic.  Is there particular installation doc one should follow?

**Comment by fjp** [comment, marked as answer]
As far as I remember I've never encountered this error so far with the [`raspicam_node`](https://github.com/UbiquityRobotics/raspicam_node) from Ubiquity Robotics. The raspicam support feature isn't fully tested/implemented yet and will be part of [Diffbot/Remo Insiders](https://ros-mobile-robots.com/insiders/).

Which exact steps did you take to get to the error you mention?

You might get it running following the steps I previously did, although in the mean time there might have been some changes on [`raspicam_node`](https://github.com/UbiquityRobotics/raspicam_node). On a fresh installation of Ubuntu Mate 20.04 I clone all required dependencies, including the `raspicam_node` (kinetic branch) by running the [`vcstool`](https://github.com/dirk-thomas/vcstool) command on Remo robot's SBC from the root of the catkin workspace and pipe in the `remo_robot.repos`  YAML file:

```
vcs import < src/diffbot/remo_robot.repos
```

Note: Alternatively, if you have most deps already you can just clone the `https://github.com/UbiquityRobotics/raspicam_node` on Remo's SBC (Raspberry Pi in your case) and make sure you have checked out the kinetic (default) branch.

Then, it should be possible to use the [`rpicamera.launch`](https://github.com/ros-mobile-robots/diffbot/blob/noetic-devel/diffbot_bringup/launch/rpicamera.launch) from the `diffbot_bringup` package to launch the `raspicam_node`. Just try to run this on Remo:

```
roslaunch diffbot_bringup rpicamera.launch
```

Then on the work/dev pc you should be able to view the camera stream, for example using [`rqt_image_view`](http://wiki.ros.org/rqt_image_view).

**Reply by questrov** [reply]
Followed the [build instructions](https://github.com/UbiquityRobotics/raspicam_node#build-intructions) but must have missed something:-/  Tried again but get `ERROR: cannot launch node of type [raspicam_node/raspicam_node]: Cannot locate node of type [raspicam_node] in package [raspicam_node]. Make sure file exists in package path and permission is set to executable (chmod +x)
` which is strange as `rospack list` finds it under `raspicam_node /home/ubuntu/catkin_ws/src/raspicam_node` the yaml file is installed where it should be and `rosdep update` says all the dependencies are there OK.  Still missing something time to give it a rest for the night:-/  Will give it a go tomorrow as you suggest above on a fresh install of Ubuntu Mate 20.04 and see what happens...

Thanks again for the help, will also check out [Diffbot/Remo insiders](https://ros-mobile-robots.com/insiders/) and join if I can.  I don't have the coding skills to contribute much but OK with the hardware, love to test new projects and seem to have a penchant for documentation.

**Reply by questrov** [reply]
Really need to quit for the night!  

Thought I would try one more thing...
`roslocate info raspicam_node`

Results in:
`Using ROS_DISTRO: noetic`
`Not found via rosdistro - falling back to information provided by rosdoc`
`error contacting http://ros.org/doc/noetic/api/raspicam_node/stack.yaml:`
`HTTP Error 404: Not Found`
`error contacting http://ros.org/doc/noetic/api/raspicam_node/manifest.yaml:`
`HTTP Error 404: Not Found`
`cannot locate information about raspicam_node`

Any clue?

**Reply by questrov** [reply]
Raspicam_node finally working OK on Pi4 with fresh install of UbuntuMate and ROS.  The late night 'missing something' was blindly copy/paste from [rasspicam_node Build Instructions](https://github.com/UbiquityRobotics/raspicam_node#build-intructions) I missed `rosdep install --from-paths src --ignore-src --rosdistro=kinetic -y` needed to replace kinetic with noetic.

**Reply by fjp** [reply]
I was just about looking into this... Thanks for posting the solution :-)
